testr/testr_test.go

```diff
@@ -29,14 +29,74 @@ func TestLogger(t *testing.T) {
 	log.V(0).Info("V(0).info")
 	log.V(1).Info("v(1).info")
 	log.Error(fmt.Errorf("error"), "error")
-	log.WithName("testing").Info("with prefix")
+	log.WithName("testing").WithValues("value", "test").Info("with prefix")
+	log.WithName("testing").Error(fmt.Errorf("error"), "with prefix")
 	Helper(log, "hello world")
 
 	log = NewWithOptions(t, Options{
 		LogTimestamp: true,
 		Verbosity:    1,
 	})
 	log.V(1).Info("v(1).info with options")
+
+	if underlier, ok := log.GetSink().(Underlier); ok {
+		if t != underlier.GetUnderlying() {
+			t.Error("invalid underlier")
+		}
+	} else {
+		t.Error("couldn't get underlier")
+	}
+}
+
+func TestLoggerInterface(t *testing.T) {
+	log := NewWithInterface(t, Options{})
+	log.Info("info")
+	log.V(0).Info("V(0).info")
+	log.V(1).Info("v(1).info")
+	log.Error(fmt.Errorf("error"), "error")
+	log.WithName("testing").WithValues("value", "test").Info("with prefix")
+	log.WithName("testing").Error(fmt.Errorf("error"), "with prefix")
+	Helper(log, "hello world")
+
+	if underlier, ok := log.GetSink().(UnderlierInterface); ok {
+		if underlierT, ok := underlier.GetUnderlying().(*testing.T); ok {
+			if t != underlierT {
+				t.Error("invalid underlier")
+			}
+		} else {
+			t.Error("couldn't get underlying *testing.T")
+		}
+	} else {
+		t.Error("couldn't get underlier")
+	}
```

Comment:
Nit: this works, but doesn't look like idiomatic Go. How about this:

```suggestion
	underlier, ok := log.GetSink().(UnderlierInterface)
	if !ok {
		t.Fatal("couldn't get underlier")
	}
	underlierT, ok := underlier.GetUnderlying().(*testing.T)
	if !ok {
		t.Fatal("couldn't get underlying *testing.T")
	}
	if t != underlierT {
		t.Error("invalid underlier")
	}
```

The same applies above and below.

Reply:
This does look better. I updated the tests to follow your suggestion.